fix: undefined behavior from unaligned reference to packed struct field in system tray

### Bug Description

In system_tray/windows.rs, the notify_icon_data function creates a reference to a field of a packed struct `NOTIFYICONDATAW`, which fails to compile.

`NOTIFYICONDATAW` is `#[repr(C, packed)]` (1-byte aligned), so its field szTip ([u16; 128]) may be unaligned. The current code accesses it via slice indexing:

```rust
let n = tip.len().min(data.szTip.len() - 1);
data.szTip[..n].copy_from_slice(&tip[..n]);
```

This implicitly creates &[u16] / &mut [u16] references to the unaligned field. Rust rejects this as a compilation error:

```shell
error[E0793]: reference to field of packed struct is unaligned
--> internal/core/items/system_tray/windows.rs:465:27
|
465 |     let n = tip.len().min(data.szTip.len() - 1);
|                           ^^^^^^^^^^
|
= note: this struct is 1-byte aligned, but the type of this field may require higher alignment
= note: creating a misaligned reference is undefined behavior (even if that reference is never dereferenced)
```

Fix: use `addr_of_mut!` to obtain a raw pointer without creating a reference, then copy via `copy_nonoverlapping`:

let sz_tip = std::ptr::addr_of_mut!(data.szTip);
let n = tip.len().min(unsafe { (*sz_tip).len() } - 1);


### Reproducible Code (if applicable)

```slint
N/A — this is a compilation error in Slint internals, not user code. Triggered whenever system tray is enabled.
```

### Environment Details

- Slint Version: master (commit c13d8ca91)
- Platform/OS: Windows (all targets — affects both Win32 and Win64)
- Programming Language: Rust
- Backend/Renderer: Skia

### Product Impact

The system tray feature is enabled via SLINT_ENABLE_EXPERIMENTAL_FEATURES=1. This blocks compilation on all both Win32 and Win64 targets.

--- a/internal/core/items/system_tray/windows.rs
+++ b/internal/core/items/system_tray/windows.rs
@@ -1,7 +1,7 @@
 // Copyright © SixtyFPS GmbH <info@slint.dev>
 // SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
 
-// cSpell: ignore BITMAPINFO BITMAPINFOHEADER BOTTOMALIGN Hotspot HSTRING ICONINFO LEFTALIGN NOTIFYICONDATAW PCWSTR RETURNCMD SZTIP TRAYICON WNDCLASSW
+// cSpell: ignore BITMAPINFO BITMAPINFOHEADER BOTTOMALIGN Hotspot HSTRING ICONINFO LEFTALIGN NOTIFYICONDATAW PCWSTR RETURNCMD TRAYICON WNDCLASSW
 //! Windows system tray backend using the `Shell_NotifyIconW` API directly.
 //!
 //! Everything here runs on the Slint event-loop thread, which on Windows is the
@@ -453,22 +453,19 @@ fn create_hicon(icon: &Image) -> Result<HICON, Error> {
 // trailing fields (`szInfo`, `szInfoTitle`, `guidItem`, …) are zero-initialized
 // via `Default::default()`, which also NUL-terminates `szTip` past the copy.
 fn notify_icon_data(hwnd: HWND, hicon: HICON, tip: &[u16]) -> NOTIFYICONDATAW {
-    let mut data = NOTIFYICONDATAW {
+    let mut buf = [0u16; 128];
+    let n = tip.len().min(buf.len() - 1);
+    buf[..n].copy_from_slice(&tip[..n]);
+    NOTIFYICONDATAW {
         cbSize: std::mem::size_of::<NOTIFYICONDATAW>() as u32,
         hWnd: hwnd,
         uID: TRAY_UID,
         uFlags: NIF_MESSAGE | NIF_ICON | NIF_TIP,
         uCallbackMessage: WM_TRAYICON,
         hIcon: hicon,
+        szTip: buf,
         ..Default::default()
-    };
-    // szTip is [u16; 128] per NOTIFYICONDATAW ABI:
-    // https://microsoft.github.io/windows-docs-rs/doc/windows/Win32/UI/Shell/struct.NOTIFYICONDATAW.html
-    const SZTIP_LEN: usize = 128;
-    let sz_tip = &raw mut data.szTip;
-    let n = tip.len().min(SZTIP_LEN - 1);
-    unsafe { std::ptr::copy_nonoverlapping(tip.as_ptr(), sz_tip.cast::<u16>(), n) };
-    data
+    }
 }
 
 fn append_menu_entry(

--- a/internal/core/items/system_tray/windows.rs
+++ b/internal/core/items/system_tray/windows.rs
@@ -462,9 +462,12 @@ fn notify_icon_data(hwnd: HWND, hicon: HICON, tip: &[u16]) -> NOTIFYICONDATAW {
         hIcon: hicon,
         ..Default::default()
     };
-    let sz_tip = std::ptr::addr_of_mut!(data.szTip);
-    let n = tip.len().min(unsafe { (*sz_tip).len() } - 1);
-    unsafe { std::ptr::copy_nonoverlapping(tip.as_ptr(), (*sz_tip).as_mut_ptr(), n) };
+    // szTip is [u16; 128] per NOTIFYICONDATAW ABI:
+    // https://microsoft.github.io/windows-docs-rs/doc/windows/Win32/UI/Shell/struct.NOTIFYICONDATAW.html
+    const SZTIP_LEN: usize = 128;
+    let sz_tip = &raw mut data.szTip;
+    let n = tip.len().min(SZTIP_LEN - 1);
+    unsafe { std::ptr::copy_nonoverlapping(tip.as_ptr(), sz_tip.cast::<u16>(), n) };
     data
 }
 

--- a/internal/core/items/system_tray/windows.rs
+++ b/internal/core/items/system_tray/windows.rs
@@ -1,7 +1,7 @@
 // Copyright © SixtyFPS GmbH <info@slint.dev>
 // SPDX-License-Identifier: GPL-3.0-only OR LicenseRef-Slint-Royalty-free-2.0 OR LicenseRef-Slint-Software-3.0
 
-// cSpell: ignore BITMAPINFO BITMAPINFOHEADER BOTTOMALIGN Hotspot HSTRING ICONINFO LEFTALIGN NOTIFYICONDATAW PCWSTR RETURNCMD TRAYICON WNDCLASSW
+// cSpell: ignore BITMAPINFO BITMAPINFOHEADER BOTTOMALIGN Hotspot HSTRING ICONINFO LEFTALIGN NOTIFYICONDATAW PCWSTR RETURNCMD SZTIP TRAYICON WNDCLASSW
 //! Windows system tray backend using the `Shell_NotifyIconW` API directly.
 //!
 //! Everything here runs on the Slint event-loop thread, which on Windows is the

--- a/internal/core/items/system_tray/windows.rs
+++ b/internal/core/items/system_tray/windows.rs
@@ -462,8 +462,9 @@ fn notify_icon_data(hwnd: HWND, hicon: HICON, tip: &[u16]) -> NOTIFYICONDATAW {
         hIcon: hicon,
         ..Default::default()
     };
-    let n = tip.len().min(data.szTip.len() - 1);
-    data.szTip[..n].copy_from_slice(&tip[..n]);
+    let sz_tip = std::ptr::addr_of_mut!(data.szTip);
+    let n = tip.len().min(unsafe { (*sz_tip).len() } - 1);
+    unsafe { std::ptr::copy_nonoverlapping(tip.as_ptr(), (*sz_tip).as_mut_ptr(), n) };
     data
 }
 

